vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java

```diff
@@ -46,4 +53,25 @@ public ISlide getSlide(String slideId) {
     public void updateSlide(Slide slide) {
         slideRepo.save((Slide) slide);
     }
+
+    @Override
+    public void deleteSlideById(String slideId, String moduleId) throws SlideDoesNotExistException {
+
+        try {
+            List<Sequence> sequences = sequenceRepo.findSequencesForModule(moduleId);
+            outer: for(Sequence sequence : sequences) {
+                for(ISlide slideIds : sequence.getSlides()) {
+                    if(slideIds.getId().equals(slideId)) {
+                        sequence.getSlides().remove(slideIds);
```

Comment:
are you sure this works and doesn't throw an exception? typically you have to use an iterator since you can't modify a list that you are currently iterating over.

Reply:
Yes, it's working. 
on cases with slide not attached to any Sequence. And in cases where slides are attached to a Sequence.

Reply:
Yes, it was throwing concurrent modification exception before. After i added continue outer, didn't throw exception and working as expected.

Reply:
that's not really a solution though (because of my next comment). an iterator should fix this though.

Reply:
okay.